classes/user/form/ResetPasswordForm.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+/**
+ * @file classes/user/form/ResetPasswordForm.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2003-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class ResetPasswordForm
+ * @ingroup user_form
+ *
+ * @brief Form to reset a user's password.
+ */
+
+namespace PKP\user\form;
+
+use APP\facades\Repo;
+use APP\template\TemplateManager;
+use PKP\form\Form;
+use PKP\form\validation\FormValidatorCSRF;
+use PKP\form\validation\FormValidatorPost;
+use PKP\form\validation\FormValidatorCustom;
+use PKP\form\validation\FormValidator;
+use PKP\form\validation\FormValidatorLength;
+use PKP\security\Validation;
+use PKP\session\SessionManager;
+
+class ResetPasswordForm extends Form {
+
+	/** @var object */
+	protected $_user;
+
+	/** @var object */
+	protected $_site;
+
+	/** @var string */
+	protected $_hash;
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct($user, $site, $hash) {
+		parent::__construct('user/userPasswordReset.tpl');
+
+		$this->_user = $user;
+		$this->_site = $site;
+		$this->_hash = $hash;
+
+		$this->addCheck(new FormValidatorLength($this, 'password', 'required', 'user.register.form.passwordLengthRestriction', '>=', $site->getMinPasswordLength()));
+		$this->addCheck(new FormValidator($this, 'password', 'required', 'user.profile.form.newPasswordRequired'));
+		$form = $this;
+		$this->addCheck(new FormValidatorCustom($this, 'password', 'required', 'user.register.form.passwordsDoNotMatch', function($password) use ($form) {
+			return $password == $form->getData('password2');
+		}));
+
+		$this->addCheck(new FormValidatorPost($this));
+		$this->addCheck(new FormValidatorCSRF($this));
+	}
+
+	/**
+	 * Get the user associated with this password
+	 */
+	public function getUser() 
+	{
+		return $this->_user;
+	}
+
+	/**
+	 * Get the site
+	 */
+	public function getSite() 
+	{
+		return $this->_site;
+	}
+
+	/**
+	 * Get the password reset hash
+	 */
+	public function getHash()
+	{
+		return $this->_hash;
+	}
+
+	/**
+	 * @copydoc Form::display
+	 */
+	public function display($request = null, $template = null) {
+		$templateMgr = TemplateManager::getManager($request);
+		$templateMgr->assign([
+			'minPasswordLength' => $this->getSite()->getMinPasswordLength(),
+			'username' =>  $this->getUser()->getUsername(),
+			'hash' => $this->getHash(),
+		]);
+		parent::display($request, $template);
+	}
+
+	/**
+	 * Assign form data to user-submitted data.
+	 */
+	public function readInputData() {
+		$this->readUserVars(array('username', 'hash', 'password', 'password2'));
+	}
+
+	/**
+	 * @copydoc Form::execute()
+	 */
+	public function execute(...$functionArgs) {
+        
+		$user = $this->getUser();
+        $user->setPassword(Validation::encryptCredentials($user->getUsername(), $this->getData('password')));
```

Comment:
Hmm, I see a mix of tab- and space-based indents here, but the automatic code formatting tools on `main` should take care of formatting these consistently. Do you see those run when you do a `git commit`?

Reply:
This still needs a look, I think!

Reply:
@asmecher this looks ok to me , see the diff at https://github.com/pkp/pkp-lib/pull/8512/files#diff-4550ee52519c4fe1edb10a6f377f4f7e92b0a728b53ef7119890c02af5d38b6d .

Reply:
It's consistent, but it uses tabs, which aren't appropriate for `main`. The `stable-3_3_0` branch (and previous) use tabs for indentation, without automatic formatting; `main` uses spaces, and should have automatic formatting applied. It looks like that's not working on your machine.

The automatic formatting works like this:
- A tool configures Git pre-commit hooks. In pkp-lib it's [Captain Hook](https://github.com/captainhookphp/captainhook) (pkp-lib), which is a Composer dependency; in the application repositories it's [Husky](https://typicode.github.io/husky/#/), which is a NPM dependency.
- For repositories using Husky, the pre-commit hook is configured in `package.json`:
    ```
    "husky": {
      "hooks": {
        "pre-commit": "lint-staged"
      }
    },
    "lint-staged": {
      "**/*.php": [
        "php lib/pkp/lib/vendor/bin/php-cs-fixer fix --config .php-cs-fixer.php --allow-risky=yes",
        "php lib/pkp/lib/vendor/bin/php-cs-fixer fix --config .php-cs-fixer.php --allow-risky=yes --dry-run"
      ]
    }
    ```
- For repositories using Captain Hook, the pre-commit hook is configured in `captainhook.json`:
    ```
    "pre-commit": {
        "enabled": true,
        "actions": [
            {
                "action": "\\CaptainHook\\App\\Hook\\PHP\\Action\\Linting",
                "options": [],
                "conditions": []
            },
            {
                "action": "./tools/pre-commit.sh",
                "options": [],
                "conditions": []
            }
        ]
    },

    ```
- Both of these use `php-cs-fixer` to apply code formatting to changed files, according to .php-cs-fixer.php and .php_cs_rules in the repo root.

You should see these run before any `git commit` is allowed to proceed.

Reply:
You are right, this is not running at least for the `pkp/lib` commit in my machine . I will check to see what causing the issue . I think it used to run but only when I make a commit from the separate checkout of `pkp-lib`, not from within any app (like `pkp/lib` folder reside within the checkout of OJS/OMP/OPS ) which is weird .

Reply:
My assumption is correct, those checks are not running when making commit from within an app checkout. Made a commit from separate checkout of `pkp-lib` and it ran. should be ok now but please check . In mean time I will fix it for the apps .